File: algorithms/1_easy/J/00771_jewels_and_stones/solution.cpp
class Solution {
public:
    int numJewelsInStones(string jewels, string stones) {
        array<bool, 128> jewels_map{};

        for (const char &jewel: jewels) {
            jewels_map[static_cast<size_t>(jewel)] = true;
        }

        int count{0};

        for (const char &stone: stones) {
            if (jewels_map[static_cast<size_t>(stone)]) {
                count += 1;
            }
        }

        return count;
    }
};

File: algorithms/1_easy/J/00771_jewels_and_stones/solution.py
class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        jewels_map = [False] * 128
        for jewel in jewels:
            jewels_map[ord(jewel)] = True

        count = 0

        for stone in stones:
            if jewels_map[ord(stone)]:
                count += 1

        return count
